### ISSUE TITLE
android.database.sqlite.SQLiteException

### ISSUE THREAD

**Issue by dmfs** [dmfs/opentasks#793]
This is currently the most frequent crash reason:
```
  at android.database.sqlite.SQLiteConnection.nativePrepareStatement (SQLiteConnection.java)
  at android.database.sqlite.SQLiteConnection.acquirePreparedStatement (SQLiteConnection.java:890)
  at android.database.sqlite.SQLiteConnection.prepare (SQLiteConnection.java:501)
  at android.database.sqlite.SQLiteSession.prepare (SQLiteSession.java:588)
  at android.database.sqlite.SQLiteProgram.<init> (SQLiteProgram.java:58)
  at android.database.sqlite.SQLiteQuery.<init> (SQLiteQuery.java:37)
  at android.database.sqlite.SQLiteDirectCursorDriver.query (SQLiteDirectCursorDriver.java:46)
  at android.database.sqlite.SQLiteDatabase.rawQueryWithFactory (SQLiteDatabase.java:1392)
  at android.database.sqlite.SQLiteQueryBuilder.query (SQLiteQueryBuilder.java:399)
  at android.database.sqlite.SQLiteQueryBuilder.query (SQLiteQueryBuilder.java:294)
  at org.dmfs.provider.tasks.TaskProvider.query (TaskProvider.java:707)
  at android.content.ContentProvider.query (ContentProvider.java:1055)
  at android.content.ContentProvider.query (ContentProvider.java:1147)
  at android.content.ContentProvider$Transport.query (ContentProvider.java:240)
  at android.content.ContentProviderClient.query (ContentProviderClient.java:154)
  at android.content.ContentProviderClient.query (ContentProviderClient.java:137)
  at android.content.ContentProviderClient.query (ContentProviderClient.java:127)
  at org.dmfs.android.contentpal.views.BaseView.rows (BaseView.java:71)
  at org.dmfs.android.contentpal.views.DelegatingView.rows (DelegatingView.java:54)
  at org.dmfs.android.contentpal.views.Sorted.rows (Sorted.java:59)
  at org.dmfs.android.contentpal.rowsets.QueryRowSet.iterator (QueryRowSet.java:71)
  at org.dmfs.android.contentpal.rowsets.QueryRowSet.iterator (QueryRowSet.java:50)
  at org.dmfs.jems.iterable.decorators.Mapped.iterator (Mapped.java:46)
  at org.dmfs.jems.iterable.composite.Diff.iterator (Diff.java:50)
  at org.dmfs.tasks.notification.TaskNotificationService.onHandleWork (TaskNotificationService.java:109)
  at android.support.v4.app.JobIntentService$CommandProcessor.doInBackground (JobIntentService.java:391)
  at android.support.v4.app.JobIntentService$CommandProcessor.doInBackground (JobIntentService.java:382)
  at android.os.AsyncTask$2.call (AsyncTask.java:333)
  at java.util.concurrent.FutureTask.run (FutureTask.java:266)
  at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1162)
  at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:636)
  at java.lang.Thread.run (Thread.java:764)
```

**Comment by dmfs** [comment]
Happens on all recent Android versions. Unfortunately there is no additional info.

**Comment by dmfs** [comment]
It happens when this query is executed (TaskProvider Line 707):
```
Cursor c = sqlBuilder.query(db, projection, selection, selectionArgs, null, null, sortOrder);
```

However, the query looks ok to me (and also works on in my tests):
```
SELECT _id, version, pinned FROM Task_View WHERE ( 1=1  AND _deleted=0) AND (( pinned = ? ) or ( _id in (?, ? )  )) ORDER BY _id
```

**Comment by dmfs** [comment]
It may be possible that for some reason the `version` column doesn't exist on the affected devices. Maybe due to some migration issue with the beta version. That would explain why there are way more reports of this for the beta channel.

**Comment by dmfs** [comment]
Hopefully fixed by #794
